time-series/time-series-api/src/main/java/com/powsybl/timeseries/ast/NodeCalcVisitor.java

```diff
@@ -52,4 +48,8 @@ public interface NodeCalcVisitor<R, A> {
     R visit(TimeSeriesNameNodeCalc nodeCalc, A arg);
 
     R visit(TimeSeriesNumNodeCalc nodeCalc, A arg);
+
+    R visit(AbstractBinaryNodeCal nodeCalc, A arg, R left, R right);
```

Comment:
You should continue with same design that previous by having in the visitor only child node and not abstracts. As you can see, as doing that you need to lot of instanceof check in all implementations.

Reply:
done